lib/drab/browser.ex

```diff
@@ -296,4 +297,191 @@ defmodule Drab.Browser do
+  * `value`   - the cookie value, can be an Elixir data structure
+        
+        
+  ### Options:
+
+  * `max-age` :
+   
+       max-age > 0    - sets the cookie max-age, in seconds
+     
+       max-age == 0   - sets a session cookie
+     
+       max-age < 0    - delete the cookie
+     
+     
+  * `path` - the cookie path, default `"/"`
+  
+  * `encode` - encodes the cookie in Base64. The encoding is needed if you want to store values with comma, semicolon, quotes, etc or Elixir data structures.   
+  
+  
+  ### Example:
+  
+        iex> Drab.Browser.set_cookie(socket, "Items", [%{id: 001, name: "foo"}, %{id: 002, name: "bar"}], max_age: 3*24*60*60, encode: true)
+        {:ok, result}
+  """
+  # @spec exec_js(Phoenix.Socket.t(), String.t(), Keyword.t()) :: result
+  def set_cookie(socket, key, value, options \\ []) do
+    # Options
+    max_age = Keyword.get(options, :max_age, 0)
+    path = Keyword.get(options, :path, "/")
+
+    # Prepare
+    encoded_value = encode_value(value, options)
+    expires = cookie_expires(socket, max_age)
+
+    # Set cookie
+    exec_js(socket, "document.cookie='#{key}=#{encoded_value}; expires=#{expires}; path=#{path};'")
+  end
+
+  @doc """
+  Exception raising version of `set_cookie/4`
+  """
+  def set_cookie!(socket, key, value, options \\ []) do
+    # Options
+    max_age = Keyword.get(options, :max_age, 0)
+    path = Keyword.get(options, :path, "/")
+
+    # Prepare
+    encoded_value = encode_value(value, options)
+    expires = cookie_expires(socket, max_age)
+
+    # Set cookie
+    exec_js!(socket, "document.cookie='#{key}=#{encoded_value}; expires=#{expires}; path=#{path};'")
+  end
+
+  @doc """
+  Delete a cookie.
+  
+  ### Parameters:
+
+  * `socket`  - the Drab socket
+  * `key`     - the cookie name
+
+  ### Example:
+  
+        iex> Drab.Browser.delete_cookie(socket, "Items")
+        {:ok, result}
+  """
+  def delete_cookie(socket, key) do
+    set_cookie(socket, key, "", max_age: -1)
+  end
+
+  @doc """
+  Exception raising version of `delete_cookie/2`
+  """
+  def delete_cookie!(socket, key) do
+    set_cookie!(socket, key, "", max_age: -1)
+  end
+
+  @doc """
+  Retrieve all cookies from browser and convert them in a list of maps.
+
+  ### Parameters
+
+  * `socket` - The Drab socket
+
+  ### Example:
+  
+        iex> Drab.Browser.cookies(socket)
+        {:ok, result}
+  """
+  def cookies(socket) do
+    socket
+    |> raw_cookies()
+    |> case do
+        {:ok, cookies} -> {:ok, extract_cookies_maps(cookies)}
+        {:error, error} -> {:error, error}
+      end
+  end
+
+  @doc """
+  Exception raising version of `cookies/1`
+  """
+  def cookies!(socket) do
+    socket
+    |> raw_cookies!()
+    |> extract_cookies_maps()
+  end
+
+  @doc """
+  Retrieve all cookies from browser.
+
+  ### Parameters
+
+  * `socket` - The Drab socket
+
+  ### Example:
+  
+        iex> Drab.Browser.raw_cookies(socket)
+        {:ok, result}
+  """
+  def raw_cookies(socket) do
```

Comment:
I don't think we need to expose `raw_cookies`. Let's keep API smallest possible.